sdks/cpp/lite/include/IParam.h

```diff
@@ -43,10 +43,7 @@ class IParam {
     /**
      * @brief return the oid of the param
      */
-    virtual const std::string& getOid() const { return oid_; }
-    
-  protected:
-    const std::string oid_;
+    virtual const std::string& getOid() const = 0;
```

Comment:
I think this should be a public method. what's the rationale for making it protected?

Reply:
I think you misread the change. Protected was removed so it is public.